collect_app/src/main/java/org/odk/collect/android/database/instances/InstanceDatabaseMigrator.java

```diff
@@ -53,18 +52,24 @@ public void onUpgrade(SQLiteDatabase db, int oldVersion) {
                 upgradeToVersion5(db);
             case 5:
                 upgradeToVersion6(db, INSTANCES_TABLE_NAME);
-                break;
             case 6:
                 upgradeToVersion7(db);
-                break;
             case 7:
+                upgradeToVersion8(db);
+                break;
+            case 8:
                 // Remember to bump the database version number in {@link org.odk.collect.android.database.DatabaseConstants}
-                // upgradeToVersion8(db);
+                // upgradeToVersion9(db);
             default:
                 Timber.i("Unknown version %d", oldVersion);
         }
     }
 
+    private void upgradeToVersion8(SQLiteDatabase db) {
+        SQLiteUtils.addColumn(db, INSTANCES_TABLE_NAME, CAN_DELETE_BEFORE_SEND, "text");
+        db.execSQL("UPDATE " + INSTANCES_TABLE_NAME + " SET " + CAN_DELETE_BEFORE_SEND + " = 'true';");
+    }
+
     public void onDowngrade(SQLiteDatabase db) {
```

Comment:
You need to update this method too. This might be not needed (see https://github.com/getodk/collect/issues/6315) but for now, let's do that.